lib/pagseguro/authorization_request/request_serializer.rb

```diff
@@ -1,29 +1,141 @@
 module PagSeguro
   class AuthorizationRequest
     class RequestSerializer
-      attr_reader :authorization_request
 
-      def initialize(authorization_request)
-        @authorization_request = authorization_request
-      end
+      attr_accessor :request
 
-      def to_params
-        params[:credentials] = authorization_request.credentials if authorization_request.credentials
-        params[:notificationURL] = authorization_request.notification_url
-        params[:redirectURL] = authorization_request.redirect_url
-        params[:permissions] = serialize_permissions(authorization_request.permissions)
-        params[:reference] = authorization_request.reference if authorization_request.reference
+      def initialize(request)
+        @request = request
+        @builder = Nokogiri::XML::Builder
+      end
 
-        params
+      def build_xml
+        build.to_xml(save_with:
+          Nokogiri::XML::Node::SaveOptions::NO_EMPTY_TAGS |
+          Nokogiri::XML::Node::SaveOptions::FORMAT
+        )
       end
 
       private
-      def params
-        @params ||= {}
+
+      attr_reader :builder
+
+      def build
+        builder.new(encoding: PagSeguro.encoding) do |xml|
+          xml.send(:authorizationRequest) {
+            xml.send(:reference, request.reference) if request.reference
+            xml.send(:redirectURL, request.redirect_url)
+            xml.send(:notificationURL, request.notification_url)
+            serialize_permissions(xml, request.permissions)
+            serialize_account(xml, request.account)
+          }
+        end
+      end
+
+      def serialize_permissions(xml, permissions)
+        return unless permissions.any?
+        xml.send(:permissions) {
+          get_permissions.each do |permission|
+            xml.send(:code, permission)
+          end
+        }
+      end
+
+      def get_permissions
```

Comment:
But has a problem, because I already have a permission as argument.
I'll need to change one or other.